Tests/SecureXPCTests/XPCEncoder/Scalar encoding tests.swift

```diff
@@ -0,0 +1,240 @@
+		try assert(.max as UInt, encodesEqualTo: xpc_uint64_create(UInt64(UInt.max)))
+	}
+	
+	func testEncodes_UInt8_as_XPCUInt64() throws {
+		try assert(.min as UInt8, encodesEqualTo: xpc_uint64_create(UInt64(UInt8.min)))
+		try assert(   0 as UInt8, encodesEqualTo: xpc_uint64_create(                0))
+		try assert( 123 as UInt8, encodesEqualTo: xpc_uint64_create(              123))
+		try assert(.max as UInt8, encodesEqualTo: xpc_uint64_create(UInt64(UInt8.max)))
+	}
+	
+	func testEncodes_UInt16_as_XPCUInt64() throws {
+		try assert(.min as UInt16, encodesEqualTo: xpc_uint64_create(UInt64(UInt16.min)))
+		try assert(   0 as UInt16, encodesEqualTo: xpc_uint64_create(                 0))
+		try assert( 123 as UInt16, encodesEqualTo: xpc_uint64_create(               123))
+		try assert(.max as UInt16, encodesEqualTo: xpc_uint64_create(UInt64(UInt16.max)))
+	}
+	
+	func testEncodes_UInt32_as_XPCUInt64() throws {
+		try assert(.min as UInt32, encodesEqualTo: xpc_uint64_create(UInt64(UInt32.min)))
+		try assert(   0 as UInt32, encodesEqualTo: xpc_uint64_create(                 0))
+		try assert( 123 as UInt32, encodesEqualTo: xpc_uint64_create(               123))
+		try assert(.max as UInt32, encodesEqualTo: xpc_uint64_create(UInt64(UInt32.max)))
+	}
+	
+	func testEncodes_UInt64_as_XPCUInt64() throws {
+		try assert(.min as UInt64, encodesEqualTo: xpc_uint64_create(UInt64(UInt64.min)))
+		try assert(   0 as UInt64, encodesEqualTo: xpc_uint64_create(                 0))
+		try assert( 123 as UInt64, encodesEqualTo: xpc_uint64_create(               123))
+		try assert(.max as UInt64, encodesEqualTo: xpc_uint64_create(UInt64(UInt64.max)))
+	}
+	
+	// MARK: Floating point numbers
+	
+	func testEncodes_Float_as_XPCDouble() throws {
+		// "Normal" values, from lowest to highest
+		try assert(-Float.infinity,
+				   encodesEqualTo: xpc_double_create(Double(-Float.infinity)))
+		try assert(-Float.greatestFiniteMagnitude,
+				   encodesEqualTo: xpc_double_create(Double(-Float.greatestFiniteMagnitude)))
+		try assert(-123.0 as Float,
+				   encodesEqualTo: xpc_double_create(-123.0))
+		try assert(-0.0 as Float,
+				   encodesEqualTo: xpc_double_create(-0.0))
+		try assert(0.0 as Float,
+				   encodesEqualTo: xpc_double_create(0.0))
+		try assert(123.0 as Float,
+				   encodesEqualTo: xpc_double_create(123.0))
+		try assert(Float.greatestFiniteMagnitude,
+				   encodesEqualTo: xpc_double_create(Double(Float.greatestFiniteMagnitude)))
+		try assert(Float.infinity,
+				   encodesEqualTo: xpc_double_create(Double(Float.infinity)))
+		
+		// NaN
+		let result = xpc_double_get_value(try encode(Float.nan))
+		XCTAssert(result.isNaN)
+	}
+	
+	func testEncodes_Float_SignalingNaN_as_XPCDouble_QuietNaN() throws {
+		let result = xpc_double_get_value(try encode(Float.signalingNaN))
+		XCTAssert(result.isNaN)
+		// IDK if this is intended or acceptable, but `Double(Float.signalingNaN).isSignalingNaN` returns `false`
```

Comment:
Huh, but it works properly when nested in an array or dictionary?

Reply:
🤷